pkgs/by-name/si/simutrans/package.nix

```diff
@@ -2,196 +2,439 @@
   lib,
   stdenv,
   fetchurl,
+  subversion,
+  fetchsvn,
+  fetchzip,
   pkg-config,
-  unzip,
   zlib,
+  zstd,
   libpng,
   bzip2,
-  SDL,
-  SDL_mixer,
-  buildEnv,
-  config,
-  runtimeShell,
+  SDL2,
+  SDL2_mixer,
+  fontconfig,
+  miniupnpc,
+  symlinkJoin,
+  makeWrapper,
+  runCommand,
+  cabextract,
 }:
-
 let
-  # Choose your "paksets" of objects, images, text, music, etc.
-  paksets = config.simutrans.paksets or "pak64 pak64.japan pak128 pak128.britain pak128.german";
-
-  result = withPaks (
-    if paksets == "*" then
-      lib.attrValues pakSpec # taking all
-    else
-      map (name: pakSpec.${name}) (lib.splitString " " paksets)
-  );
-
-  ver1 = "121";
-  ver2 = "0";
-  ver3 = "";
-  version = "${ver1}.${ver2}${lib.optionalString (ver3 != "") ".${ver3}"}";
-  ver_dash = "${ver1}-${ver2}${lib.optionalString (ver3 != "") "-${ver3}"}";
-
-  binary_src = fetchurl {
-    url = "mirror://sourceforge/simutrans/simutrans/${ver_dash}/simutrans-src-${ver_dash}.zip";
-    sha256 = "1f463r6kr5ig0zd3mncc74k93xbjywsq3d06j5r17831jyc9bzb9";
+  fetchcab =
+    args:
+    let
+      src = fetchurl args;
+    in
+    runCommand "source-cab" { nativeBuildInputs = [ cabextract ]; } ''
+      cabextract ${src} -d $out
+    '';
+
+  fetchzip' = defaults: args: fetchzip (defaults // args);
```

Comment:
Don't we need to use override to apply the args to fetchzip?

Reply:
need? no, this works fine.

could? almost certainly. i actually never thought about the fact that `fetchzip` is just `stdenv.mkDerivation` and that it will naturally support overrides.

Reply:
wait no, i can't do that at all.

this part is overridable:

https://github.com/NixOS/nixpkgs/blob/0a81508bd9a411b1c08eb241f990912a3ff6f874/pkgs/build-support/fetchzip/default.nix#L8-L14

but this part is just a normal function, not `mkOverridable`:

https://github.com/NixOS/nixpkgs/blob/0a81508bd9a411b1c08eb241f990912a3ff6f874/pkgs/build-support/fetchzip/default.nix#L16-L33

so, what i do here with my `fetchzip'` is the only way to do it i think? `fetchurl` is basically the same, and `fetchzip` is essentially implemented as `fetchurl (defaults // args)` as well.

https://github.com/NixOS/nixpkgs/blob/0a81508bd9a411b1c08eb241f990912a3ff6f874/pkgs/build-support/fetchzip/default.nix#L102

Reply:
> need? no, this works fine.

Oh, I missed the (). With them it works, without it doesn't. Probably fine to keep. If we had deep attrs we should probably use lib.recursiveUpdate.